Assets/SEE/UI/Window/MetricWindow.cs

```diff
@@ -0,0 +1,215 @@
+                foreach (string ele in searchList)
+                {
+                    searchableObjects.TryGetValue(ele, out GameObject activeObject);
+                    activeObject.SetActive(true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the attributes of all <paramref name="gameObjects"/> whose name match the
+        /// <paramref name="query"/> using a fuzzy search.
+        /// </summary>
+        /// <param name="query"> the search query (part of an attribute name)</param>
+        /// <param name="gameObjects"> the game objects to search through</param>
+        /// <param name="cutoff"> representing the cutoff score for relevance </param>
+        /// <returns> the attributes whose names match the <paramref name="query"/> </returns>
+        public IEnumerable<string> Search(string query, GameObject[] gameObjects, int limit = 15, int cutoff = 40)
+        {
+            List<string> attributesList = new();
+            foreach (GameObject obj in gameObjects)
+            {
+                attributesList.Add(AttributeName(obj));
+            }
+            IEnumerable<(int score, string attribute)> searchResults
+                = Process.ExtractTop(query, attributesList, limit: limit, cutoff: cutoff).Select(x => (x.Score, x.Value));
+            searchResults = searchResults.OrderByDescending(x => x.score);
+
+            return searchResults.Select(x => x.attribute);
+        }
+
+        /// <summary>
+        /// Returns the name of a node attribute stored in the first child of the <paramref name="metricRow"/>.
+        /// Parameter <paramref name="metricRow"/> is assumed to represent a row in the metric window providing
+        /// the name and value of a node attribute (metric).
+        /// </summary>
+        /// <param name="metricRow">a game object representing a pair of an attribute name and an attribute value</param>
+        /// <returns>name of the node attribute</returns>
+        private static string AttributeName(GameObject metricRow)
+        {
+            return Attribute(metricRow).text;
+        }
+
+        /// <summary>
+        /// Returns the TMP container of <paramref name="metricRow"/> holding the attribute name.
+        /// Parameter <paramref name="metricRow"/> is assumed to represent a row in the metric window providing
+        /// the name and value of a node attribute (metric).
+        /// </summary>
+        /// <param name="metricRow">a game object representing a pair of an attribute name and an attribute value</param>
+        /// <returns>the TMP holding the attribute name</returns>
+        /// <remarks>Assumes that the attribute name is stored in the first child of the metric row.</remarks>
+        private static TextMeshProUGUI Attribute(GameObject metricRow)
+        {
+            return metricRow.transform.GetChild(0).GetComponent<TextMeshProUGUI>();
```

Comment:
I recommend using `MustGetComponent` here, as this leads to more descriptive error messages in case we ever change the prefab:
```suggestion
            return metricRow.transform.GetChild(0).MustGetComponent<TextMeshProUGUI>();
```